python/lib/core/dmod/core/data_domain_detectors.py

```diff
@@ -0,0 +1,458 @@
+            Indication of whether :method:`detect` should short circuit and return the 1st successful detection, rather
+            than try all subclasses and risk multiple detections, and thus an error condition (default: ``False``).
+        type_sort_func: Optional[Callable[[Type[ItemDataDomainDetector]], Any]]
+            Optional function necessary for calls to usage of the built-in ``sorted`` function to sort detector
+            subclasses during various instance operations, and serving as the ``key`` argument to ``sorted``; note that
+            sorting is performed in such places IFF this is validly set, as the subclass themselves - i.e., the
+            :class:`type` objects - do not implement `<`.
+        kwargs
+
+        Keyword Args
+        ------------
+        item: DataItem
+            The data item for which a domain will be detected.
+        item_name: Optional[str]
+            The name for the item, which includes important domain metadata in some situations.
+        decode_format: str
+            The decoder format when decoding byte strings (``utf-8`` by default).
+        """
+        super().__init__(**kwargs)
+        self._detector_types: Set[Type[ItemDataDomainDetector]] = set(detector_types)
+        if len(self._detector_types) == 0:
+            raise ValueError(f"{self.__class__.__name__} received empty collection of detector subclasses during init")
+        self._short_on_success: bool = short_on_success
+        self._type_sort_func: Optional[Callable[[Type[ItemDataDomainDetector]], Any]] = type_sort_func
+
+    def _try_detection(self, detector_type: Type[ItemDataDomainDetector]) -> Optional[DataDomain]:
+        """
+        Initialize an instance of the given detector subclass and attempt domain detection using it.
+
+        Initialize an instance of the given detector subclass, using only "vanilla" init params inherent of
+        :class:`ItemDataDomainDetector`: ``item``, ``item_name``, and ``decode_format``.  Then call that instance's
+        :method:`ItemDataDomainDetector.detect` method, passing no extra keyword args.  If ``detect`` succeeds, return
+        the resulting :class:`DataDomain`, but if it fails, catch the raised exception and return ``None``.
+
+        Parameters
+        ----------
+        detector_type: Type[ItemDataDomainDetector]
+            The subclass of :class:`ItemDataDomainDetector` to use for detection.
+
+        Returns
+        -------
+        Optional[DataDomain]
+            The domain, if it could be detected by the sub-instance; otherwise ``None``.
+
+        Raises
+        ------
+        TypeError
+            Raised if an associated subclass type requires additional init params beyond ``item``, ``item_name``, and
+            ``decode_format``.
+        """
+        detector = detector_type(item=self._item, item_name=self._item_name, decode_format=self._decode_format)
+        try:
+            return detector.detect()
+        except:
```

Comment:
Eh, I go back and forth a little, but I went ahead and did this.  On some level, I don't like it.  It clutters the code, and (all things being equal) I would much rather just get out the debugger.  But sometimes that isn't straightforward to do for one reason or another, and maybe this could save some time and effort in the future.